clients/destination.go

```diff
@@ -264,6 +264,10 @@ func (c *DestinationClient) Write(ctx context.Context, source string, syncTime t
 			Source:    source,
 			Timestamp: timestamppb.New(syncTime),
 		}); err != nil {
+			if err == io.EOF {
```

Comment:
Im not sure i understand what this solved. give in othercase this just returns an error and doesn't write to the channel while with break it actually sends `CloseAndRecv`.

Reply:
`Write` works in stream mode (it's not a synchronous RPC), so if we want the correct response/error from the write request we need to call `RecvMsg` which is only done in `CloseAndRecv` (and only once actually).

So `SendMsg` will never return application level error as it doesn't wait for the message to be received by the server.
Technically I think we should call `RecvMsg` for each `SendMsg` per the docs:
> SendMsg does not wait until the message is received by the server. An untimely stream closure may result in lost messages. To ensure delivery, users should ensure the RPC completed successfully using RecvMsg.